wandb/apis/public/api.py

```diff
@@ -1172,14 +1161,31 @@ def artifact(self, name, type=None):
             # set entity to match the settings since in above code it was potentially set to an org
             entity = self.settings["entity"] or self.default_entity
         artifact = wandb.Artifact._from_name(
-            entity, project, artifact_name, self.client, organization
+            entity, project, artifact_name, self.client, organization, enable_tracking
         )
         if type is not None and artifact.type != type:
             raise ValueError(
                 f"type {type} specified but this artifact is of type {artifact.type}"
             )
         return artifact
 
+    @normalize_exceptions
+    def artifact(self, name, type=None):
```

Comment:
would be nice to have type annotation here as well

Reply:
Sure. I guess I didn't update `artifact()`'s decostrings/definition much since I was focused on writing the internal helper. Thanks for the suggestions! Will incorporate these changes.

Reply:
yeah, assumed if you already touching this code, would be good to have it done properly! thanks